linkerd/app/outbound/src/discover/tests.rs

```diff
@@ -190,14 +185,21 @@ async fn no_profiles_when_outside_search_nets() {
 
     // Create a profile stack that uses the tracked inner stack, configured to never actually do
     // profile resolutions for the IP being tested.
-
     let cfg = {
         let mut cfg = default_config();
         // Permits resolutions for only 192.0.2.66/32.
         cfg.allow_discovery =
             AddrMatch::new(None, Some(IpNet::from(IpAddr::from([192, 0, 2, 66]))));
         cfg
     };
+    let profiles = {
+        // XXX we should assert that the resolver isn't even invoked, but the mocked resolver
+        // doesn't support that right now. So, instead, we return a profile for resolutions to
+        // and assert (below) that no profile is provided.
+        let resolver = support::profile::resolver().profile(addr, profiles::Profile::default());
+        super::WithAllowlist::new(resolver, cfg.allow_discovery.clone())
```

Comment:
Why do we need the allowlist here. What are we testing with it?

Reply:
the only thing that this test is testing is...whether the allowlist works. i just removed the test.